app/utils/reducerInjectors.js

```diff
@@ -0,0 +1,57 @@
+import invariant from 'invariant';
+import isEmpty from 'lodash/isEmpty';
+import isFunction from 'lodash/isFunction';
+import isString from 'lodash/isString';
+
+import checkStore from './checkStore';
+import createReducer from '../reducers';
+
+const allowedModes = ['restart-on-remount', 'daemon'];
+
+export function injectReducerFactory(store, isValid) {
+  return function injectReducer(name, reducer) {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name) && isFunction(reducer),
+      '(app/utils...) injectReducer: Expected `reducer` to be a reducer function'
+    );
+
+    // Check `store.injectedReducers[name] === reducer` for hot reloading when a name is the same but a reducer is different
+    // otherwise daemon reducers won't hot reload
+    if (Reflect.has(store.injectedReducers, name) && store.injectedReducers[name] === reducer) return;
+
+    store.injectedReducers[name] = reducer; // eslint-disable-line no-param-reassign
+    store.replaceReducer(createReducer(store.injectedReducers));
+  };
+}
+
+export function ejectReducerFactory(store, isValid) {
+  return function ejectReducer(name, mode = 'restart-on-remount') {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name),
+      '(app/utils...) injectReducer: Expected `name` and `mode` to be a non empty string with a correct value'
+    );
+
+    invariant(
+      isString(mode) && allowedModes.includes(mode),
+      `(app/utils...) injectReducer: Expected \`mode\` to be one of: ${allowedModes.join(', ')}`
+    );
+
+    if (mode !== 'daemon') {
+      Reflect.deleteProperty(store.injectedReducers, name);
+      store.replaceReducer(createReducer(store.injectedReducers));
+    }
+  };
+}
+
+export default function getInjectors(store) {
```

Comment:
@justingreenberg, do you mean to hide those injectors in the file with a decorator and make them private? And rename the file to `withInjectors`? About renaming, I thought the convention was to give a decorator a name that starts with a verb like 'connect', 'inject', 'fetch', etc and then call a variable `with ...` when we use the decorator, from apollo docs:

 ```js
const withData = graphql(COMMENT_QUERY, {
    name: 'comments',
    options: ({ params }) => ({
        variables: {
            repoName: `${params.org}/${params.repoName}`
        },
    })
});
export const CommentsPageWithData = withData(CommentsPage);
```

Do you also want to combine two decorators into one: `withInjectors` or `withSagaInjectors`? 

> i'm not sure we should recommend using context to access store

I meant If they are not satisfied with our decorator they create their own and use `injectSaga` inside and of course they would need to access the store just like we do.